src/main/java/com/arextest/common/filter/ContentFilter.java

```diff
@@ -0,0 +1,123 @@
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Cache the response and record the size of the request and response
+ * created by xinyuan_wang on 2023/12/25
+ */
+@Slf4j
+public class ContentFilter implements Filter {
+  private final List<MetricListener> metricListeners;
+  private final boolean recordPayloadWithMetric;
+  private static Method method;
+  public ContentFilter(List<MetricListener> metricListeners, boolean recordPayloadWithMetric) {
+    this.metricListeners = metricListeners;
+    this.recordPayloadWithMetric = recordPayloadWithMetric;
+  }
+
+  @Override
+  public void init(FilterConfig filterConfig) {
+    // do nothing
+  }
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
+    if (CollectionUtils.isEmpty(metricListeners)) {
+      chain.doFilter(request, response);
+      return;
+    }
+    if (skipMetric(((HttpServletRequest) request).getMethod())) {
+      chain.doFilter(request, response);
+      return;
+    }
+    HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+    chain.doFilter(httpServletRequest, response);
+    logResponseSize(httpServletRequest, response);
+  }
+
+  private void logResponseSize(HttpServletRequest httpServletRequest,
+      ServletResponse servletResponse) {
+    String clientApp = httpServletRequest.getHeader(CLIENT_APP_HEADER);
+    String category = httpServletRequest.getHeader(CATEGORY_TYPE_HEADER);
+    String requestURI = httpServletRequest.getRequestURI();
+    long requestLength = httpServletRequest.getContentLengthLong();
+    long responseLength = getResponseLength(servletResponse);
+    recordPayloadInfo(clientApp, category, requestURI, requestLength, responseLength);
+  }
+
+  private long getResponseLength(ServletResponse servletResponse) {
+    Class<? extends ServletResponse> responseClass = servletResponse.getClass();
+    String responseClassName = responseClass.getName();
+    if ("org.mortbay.jetty.Response".equals(responseClassName)) {
+      // jetty
+      return getResponseLengthByMethod(servletResponse, "getContentCount");
+    } else if ("org.apache.catalina.connector.ResponseFacade".equals(responseClass.getName())) {
+      //tomcat 7+
+      return getResponseLengthByMethod(servletResponse, "getContentWritten");
+    } else {
+      return 0L;
+    }
+  }
+
+  private long getResponseLengthByMethod(ServletResponse servletResponse, String methodName) {
+    if (method == null) {
```

Comment:
if previous cached `method` is `getContentCount` method, but this is call `getContentWritten` method, will there be no problem? if this situation does not exist, it is best to add a comment.

Reply:
The getContentWritten method returns the number of bytes that have been written to the response body, including all contents of the response header and response body. However, it does not include the number of bytes in the response headers, since response headers are sent before writing the response body in Jetty. Therefore, if you want to calculate the size of the entire HTTP response, you should use the getContentCount method, which returns the size of the entire HTTP response, including all contents of the response headers and response body.